src/arch/riscv/pagetable_walker.cc

```diff
@@ -490,8 +490,19 @@ Walker::WalkerState::recvPacket(PacketPtr pkt)
             Addr paddr = walker->tlb->translateWithTLB(vaddr, satp.asid, mode);
             req->setPaddr(paddr);
             walker->pma->check(req);
+
+            // If we are not in the m-mode, check pmp
+            // permissions
+            if (pmode != PrivilegeMode::PRV_M) {
+                bool pass = walker->pmp->pmpCheck(req, mode);
+                if (!pass) {
```

Comment:
I don't love this variable name. I'd just do:
```suggestion
                if (!walker->pmp->pmpCheck(req, mode)) {
```

Reply:
updated.